test/yagna/module/payments/test_zero_amount_txs.py

```diff
@@ -0,0 +1,142 @@
+    demand_constraints: str,
+    exe_script: dict,
+    payment_id_pool: PaymentIdPool,
+    runner: Runner,
+    task_package_template: str,
+):
+    """Test successful flow requesting WASM tasks with goth REST API client."""
+
+    topology = _topology(assets_path, payment_id_pool)
+
+    async with runner(topology):
+        requestor = runner.get_probes(probe_type=RequestorProbeWithApiSteps)[0]
+        provider = runner.get_probes(probe_type=ProviderProbeWithLogSteps)[0]
+
+        # Market
+
+        await provider.wait_for_offer_subscribed()
+
+        task_package = task_package_template.format(
+            web_server_addr=runner.host_address, web_server_port=runner.web_server_port
+        )
+
+        subscription_id, demand = await requestor.subscribe_template_demand(
+            task_package, demand_constraints
+        )
+
+        proposal = (
+            await requestor.wait_for_proposals(
+                subscription_id,
+                [provider],
+                lambda proposal: (
+                    proposal.properties.get("golem.runtime.name") == "wasmtime"
+                ),
+            )
+        )[0]
+        logger.info("Collected the proposal")
+
+        logger.info("Processing proposal from %s", provider.name)
+
+        # HINT: According to @kubam this is needed.
+        counterproposal_id = await requestor.counter_proposal(
+            subscription_id, demand, proposal
+        )
+        await provider.wait_for_proposal_accepted()
+
+        new_proposals = await requestor.wait_for_proposals(subscription_id, (provider,))
+        new_proposal = new_proposals[0]
+        assert new_proposal.prev_proposal_id == counterproposal_id
+
+        # Here: Agreement
+        agreement_id = await requestor.create_agreement(new_proposal)
+        await requestor.confirm_agreement(agreement_id)
+        await provider.wait_for_agreement_approved()
+
+        await requestor.unsubscribe_demand(subscription_id)
+        logger.info("Got agreement")
+
+        #  Zero-amount invoice is issued when agreement is terminated without activity
+        await requestor.wait_for_approval(agreement_id)
+        await requestor.terminate_agreement(agreement_id, None)
+
+        # Payment
+
+        await provider.wait_for_invoice_sent()
+        invoices = await requestor.gather_invoices(agreement_id)
+        await requestor.pay_invoices(invoices)
+        await provider.wait_for_invoice_paid()
+
+        # verify requestor's invoice is settled
+        invoice = (await requestor.gather_invoices(agreement_id))[0]
+        assert invoice.amount == "0"
+        assert invoice.status == InvoiceStatus.SETTLED
+
+        # verify the provider invoice is settled
+        # TODO: Expose PaymentOperationMixin from Provider-based probe
+        # pseudo-code:
+        # invoice2 = (await providers.gather_invoices(agreement_id))[0]
+        # assert invoice2.amount == "0"
+        # assert invoice2.status == InvoiceStatus.SETTLED
```

Comment:
Let's not leave this block hanging here, I've added an issue (https://github.com/golemfactory/goth/issues/433) to track progress.